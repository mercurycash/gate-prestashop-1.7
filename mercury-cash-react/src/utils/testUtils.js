import { render } from "@testing-library/react";
import { Provider } from "outstated";
import { store } from "../state/state";

export const customRender = (ui) => {
  return render(
    <Provider stores={ [ store ] }>{ ui }</Provider>
  )
}
