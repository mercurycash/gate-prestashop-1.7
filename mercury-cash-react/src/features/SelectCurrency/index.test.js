import { SelectCurrency } from "./index";
import { customRender } from "../../utils/testUtils";

describe('<SelectCurrency/>', () => {
  it('should render component and create snapshot', () => {
    const { container } = customRender(<SelectCurrency/>)
    expect(container).toMatchSnapshot()
  })
})