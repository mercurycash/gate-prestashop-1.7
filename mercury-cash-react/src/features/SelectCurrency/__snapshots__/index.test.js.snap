// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectCurrency/> should render component and create snapshot 1`] = `
<div>
  <h3
    class="mercury-cash__title mercury-cash__title--big"
  >
    payWithCryptoCurrency
  </h3>
  <h4
    class="mercury-cash__subtitle"
  >
    selectCrypto
  </h4>
  <button
    class="mercury-cash__btn-reset mercury-cash__btn-crypto"
  >
    Bitcoin
    <svg>
      icon-btc.svg
    </svg>
  </button>
  <button
    class="mercury-cash__btn-reset mercury-cash__btn-crypto"
  >
    Ethereum
    <svg>
      icon-eth.svg
    </svg>
  </button>
  <button
    class="mercury-cash__btn-reset mercury-cash__btn-crypto"
  >
    Dash
    <svg>
      icon-dash.svg
    </svg>
  </button>
</div>
`;
